deployment/build-s3-dist.sh

```diff
@@ -1,100 +1,68 @@
 #!/bin/bash
 #
-# This assumes all of the OS-level configuration has been completed and git repo has already been cloned
+# The script is for aws-solutions internal purposes only
 #
 # This script should be run from the repo's deployment directory
 # cd deployment
-# ./build-s3-dist.sh source-bucket-base-name trademarked-solution-name version-code
+# ./build-s3-dist.sh trademarked-solution-name source-bucket-base-name version-code
 #
-# Paramenters:
+# For example: ./build-s3-dist.sh my-solution solutions v1.0.0
+# Parameters:
 #  - source-bucket-base-name: Name for the S3 bucket location where the template will source the Lambda
 #    code from. The template will append '-[region_name]' to this bucket name.
-#    For example: ./build-s3-dist.sh solutions my-solution v1.0.0
 #    The template will then expect the source code to be located in the solutions-[region_name] bucket
-#
 #  - trademarked-solution-name: name of the solution for consistency
-#
 #  - version-code: version of the package
 
+[ "$DEBUG" == 'true' ] && set -x
+set -e
+
 # Check to see if input has been provided:
 if [ -z "$1" ] || [ -z "$2" ] || [ -z "$3" ]; then
     echo "Please provide the base source bucket name, trademark approved solution name and version where the lambda code will eventually reside."
-    echo "For example: ./build-s3-dist.sh solutions trademarked-solution-name v1.0.0"
+    echo "For example: ./build-s3-dist.sh trademarked-solution-name solutions v1.0.0"
     exit 1
 fi
 
-# Exit immediately if a command exits with a non-zero status.
-set -e
+function headline(){
+  echo "------------------------------------------------------------------------------"
+  echo "$1"
+  echo "------------------------------------------------------------------------------"
+}
 
-# Get reference for all important folders
+headline "[Init] Setting up paths and variables"
+solution_name="$1"
+asset_bucket="$2"
+solution_version="$3"
 template_dir="$PWD"
```

Comment:
nit: I prefer when scripts don't only work when run from the correct directory. You can get paths relative to the file like this: `local root_dir=$(dirname "$(cd -P -- "$(dirname "$0")" && pwd -P)")`